MetaMask/metamask-extension
# Fix account selectors when balances are missing

## Explanation

Some of the account selectors we use would return an empty set of accounts if the `AccountTracker` state was missing. This resulted in UI crashes when trying to access the current selected account.

The selectors have been updated to use the `identities` as the source- of-truth for the full set of accounts. This ensures that even if the balances are missing, each account will at least be represented by an empty object.

## Manual Testing Steps

1. Load v10.33.1
2. On mainnet, create a transaction but don't confirm
3. In the background console, throttle the network to be extremely slow (e.g. 1 kb upload, 1 kb download, latency 1000)
4. yarn start on **v10.34.2**, and then upgrade the install from step 1 to that version. (Important: make sure that you don't have a password in .metamaskrc before building)
5. open the extension, log in
6. Loading spinner will appear, let it run a few seconds
7. go to background console, change the network throttling to the "Slow 3G" option
8. open the extension again
9. see the error screen

(If you do the above and replace v10.34.2 in step 4 with this PRs branch, you won't see the error and everything should work as expected)

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
